scripts/communityModules/chat/FloofChat.html

```diff
@@ -0,0 +1,596 @@
+                        break;
+                    case ext == "iframe":
+                        elContent = elContent.replace(this, "<br/><iframe class=\"z-depth-2\" src='" + this + "'width=\"440\" height=\"248\" frameborder=\"0\"></iframe>");
+                        break;
+                    case ext == "webm":
+                        elContent = elContent.replace(this, "<br/><video controls class=\"z-depth-2 responsive\"><source src='" + this + "' type='video/" + ext + "'></video><br/><a href=\"javascript:gotoClipboard('" + this + "');\">" + this + "</a>");
+                        break;
+                    case protocol === "HIFI":
+                    case protocol === "hifi":
+                        elContent = elContent.replace(this, "<br/><a href=\"javascript:gotoHiFi('" + this + "');\">" + this + "</a>");
+                        break;
+                    case !!this.match(/^(?:https?:\/\/)?(?:www\.)?youtube\.com\/watch\?(?=.*v=((\w|-){11}))(?:\S+)?$/):
+                        var youtubeMatch = this.match(/^.*(youtu.be\/|v\/|u\/\w\/|embed\/|watch\?v=|\&v=)([^#\&\?]*).*/);
+                        if (youtubeMatch && youtubeMatch[2].length == 11 && this.match(/^(?:https?:\/\/)?(?:www\.)?youtube\.com\/watch\?(?=.*v=((\w|-){11}))(?:\S+)?$/)) {
+                            elContent = "<br/><iframe class=\"z-depth-2\" width='420' height='236' src='https://www.youtube.com/embed/" + youtubeMatch[2] + "' frameborder='0'></iframe><br/><a href=\"javascript:gotoURL('" + this + "');\">" + this + "</a>";
+                            break;
+                        }
+                    // else fall through to default
+                    default:
+                        elContent = elContent.replace(this, "<br/><a href=\"javascript:gotoURL('" + this + "');\">" + this + "</a>");
+                        break;
+                }
+
+            });
+            return elContent;
+        }
+
+        // Find any mentions (e.g. @andrs) and turn them into links that
+        // refer to the appropriate social profile (e.g. twitter or instagram).
+        function _linkifyMentions(text, baseUrl) {
+            return text.replace(/(^|\s|\(|>)@(\w+)/g, "$1<a href='" + baseUrl + "$2' target='_blank'>@$2</a>");
+        }
+
+        // Find any hashtags (e.g. #linkyrocks) and turn them into links that refer
+        // to the appropriate social profile.
+        function _linkifyHashtags(text, links) {
+            // If there is no search URL for a hashtag, there isn't much we can do
+            if (links.hashtagSearchUrl === null) return text;
+            return text.replace(/(^|\s|\(|>)#((\w|[\u00A1-\uFFFF])+)/g, "$1<a href='" + links.baseUrl + links.hashtagSearchUrl + "$2' target='_blank'>#$2</a>");
+        }
+
+    }(jQuery));
+
+    //End George Function
+
+    function replaceFormatting(text) {
```

Comment:
Code duplication with FloofChat.js. Can a single source be shared?

Reply:
No cause both are slightly different I believe